src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs

```diff
@@ -224,5 +256,20 @@ private SignPackageRequest GenerateSignPackageRequest(SignArgs signArgs, X509Cer
 
             return matchingCertCollection[0];
         }
+
+        private static X509Certificate2Collection FilterMatchingCertificates(X509Certificate2Collection matchingCollection)
```

Comment:
:wrench: It looks like this method removes all certs that are not marked for code signing, I suggest giving it a name that implies that. `Matching` didn't convey that to me when I read through this initially.